modules/ibmz-configure-devices-mco.adoc

```diff
@@ -0,0 +1,263 @@
+----
+<1> The role you have defined in the machine config file.
+<2> The Base64 encoded string that you have generated in the previous step.
+<3> The path where the udev rule is located.
+
+[id="configuring-dasd"]
+== Configuring DASD
+
+The following is an example of how to configure a DASD device by adding a udev rule.
+
+.Procedure
+
+. Take the following sample udev rule `41-dasd-eckd-0.0.4444.rules`:
++
+[source,terminal]
+----
+ACTION=="add", SUBSYSTEM=="ccw", KERNEL=="0.0.4444", DRIVER=="dasd-eckd", GOTO="cfg_dasd_eckd_0.0.4444"
+ACTION=="add", SUBSYSTEM=="drivers", KERNEL=="dasd-eckd", TEST=="[ccw/0.0.4444]", GOTO="cfg_dasd_eckd_0.0.4444"
+GOTO="end_dasd_eckd_0.0.4444"
+
+LABEL="cfg_dasd_eckd_0.0.4444"
+ATTR{[ccw/0.0.4444]online}="1"
+
+LABEL="end_dasd_eckd_0.0.4444"
+----
+
+. Convert the rule to Base64 encoded, by running the following command:
++
+[source,terminal]
+----
+$ base64 /path/to/file/
+----
+
+. Copy the following MCO sample profile into a YAML file:
++
+[source,yaml]
+----
+apiVersion: machineconfiguration.openshift.io/v1 
+kind: MachineConfig 
+metadata:
+   labels:
+     machineconfiguration.openshift.io/role: worker0 <1>
+   name: 99-worker0-devices
+spec:
+   config:
+     ignition:
+       version: 3.2.0
+     storage:
+       files:
+       - contents:
+           source: data:text/plain;base64,<encoded_base64_string> <2>
+         filesystem: root
+         mode: 420
+         path: /etc/udev/rules.d/41-dasd-eckd-0.0.4444.rules <3>
+----
+<1> The role you have defined in the machine config file.
+<2> The Base64 encoded string that you have generated in the previous step.
+<3> The path where the udev rule is located.
+
+[id="configuring-qeth"]
+== Configuring qeth
+
+The following is an example of how to configure a qeth device by adding a udev rule.
+
+.Procedure
+
+. Take the following sample udev rule `41-qeth-0.0.1000.rules`:
++
+[source,terminal]
+----
+ACTION=="add", SUBSYSTEM=="drivers", KERNEL=="qeth", GOTO="group_qeth_0.0.1000"
+ACTION=="add", SUBSYSTEM=="ccw", KERNEL=="0.0.1000", DRIVER=="qeth", GOTO="group_qeth_0.0.1000"
+ACTION=="add", SUBSYSTEM=="ccw", KERNEL=="0.0.1001", DRIVER=="qeth", GOTO="group_qeth_0.0.1000"
+ACTION=="add", SUBSYSTEM=="ccw", KERNEL=="0.0.1002", DRIVER=="qeth", GOTO="group_qeth_0.0.1000"
+ACTION=="add", SUBSYSTEM=="ccwgroup", KERNEL=="0.0.1000", DRIVER=="qeth", GOTO="cfg_qeth_0.0.1000"
+GOTO="end_qeth_0.0.1000"
+
+LABEL="group_qeth_0.0.1000"
+TEST=="[ccwgroup/0.0.1000]", GOTO="end_qeth_0.0.1000"
+TEST!="[ccw/0.0.1000]", GOTO="end_qeth_0.0.1000"
+TEST!="[ccw/0.0.1001]", GOTO="end_qeth_0.0.1000"
+TEST!="[ccw/0.0.1002]", GOTO="end_qeth_0.0.1000"
+ATTR{[drivers/ccwgroup:qeth]group}="0.0.1000,0.0.1001,0.0.1002"
+GOTO="end_qeth_0.0.1000"
+
+LABEL="cfg_qeth_0.0.1000"
+ATTR{[ccwgroup/0.0.1000]online}="1"
+
+LABEL="end_qeth_0.0.1000"
+----
+
+. Convert the rule to Base64 encoded, by running the following command:
```

Comment:
Comma could be removed here, eg: 

. Convert the rule to Base64 encoded by running the following command: